cmd/nodepool/openstack/create.go

```diff
@@ -65,6 +95,10 @@ func (o *RawOpenStackPlatformCreateOptions) Validate() (*ValidatedOpenStackPlatf
 		return nil, fmt.Errorf("image name is required")
 	}
 
+	if err := validateAdditionalPorts(o.AdditionalPorts); err != nil {
+		return nil, err
+	}
```

Comment:
Notice how by doing this both here and in Complete() we end up parsing the arguments twice.

Why don't we parse them just once, here, and replace AdditionalPorts in `ValidatedOpenStackPlatformCreateOptions` with a slice of parsed structs. Then we don't have to do it again in Complete(), we can just use the validated result.